app/models/user.rb

```diff
@@ -927,8 +930,20 @@ def password_validator
   end
 
   def confirm_password?(password)
-    return false unless password_hash && salt
-    self.password_hash == hash_password(password, salt)
+    return false unless password_hash && salt && password_algorithm
+    confirmed = self.password_hash == hash_password(password, salt, password_algorithm)
+
+    if confirmed && persisted? && password_algorithm != TARGET_PASSWORD_ALGORITHM
+      # Regenerate password_hash with new algorithm and persist
+      salt = SecureRandom.hex(16)
```

Comment:
I am not sure how generic we wish this implementation to be, but I believe the salt generation should not be fixed to a length of 256 bits. This is not a problem for now anyway.

Reply:
I'll move '16' into a constant `User:: so that it's less magic 👍